smartsim/database/orchestrator.py

```diff
@@ -85,8 +85,8 @@ def __init__(
 
         :param port: TCP/IP port, defaults to 6379
         :type port: int, optional
-        :param interface: network interface, defaults to "lo"
-        :type interface: str, optional
+        :param interface: network interface(s), defaults to "lo"
+        :type interface: str, list[str], optional
```

Comment:
minor doc thing: earlier in the docstrings, the type "string or list of strings" was conveyed as `str | list[str]`. Should we do that here as well/update the earlier one to be `str, list[str]` just to be consistent?

Reply:
Should it be `Optional[str] | Optional[list[str]]`?

Reply:
Good question! The "proper" (i.e. mypy accepted) type hint would be `Optional[Union[str, list[str]]]` in py<=3.9 syntax or `str | list[str] | None` in py>=3.10 syntax

But this also brings up our somewhat inconsistent use of `, optional`. For the most part, when `, optional` is provided in the `:type:` tag we usually imply that a type of `None` is accepted and we simply do not annotate the `None` in the docstr `:type:`. I swear I've seen other packages do something similar, but cannot seem to recall any offhand.

Any preference between `:type interface: str | list[str] | None, optional` or `:type interface: str | list[str], optional`? I honestly could be swayed either way...